File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> count (nums.begin() , nums.end());
        int longest = 0; 
        for(int num : count)
        {
            if(count.find(num -1) == count.end())
            {
                int length = 1;
                while(count.find(num + length) != count.end())
                {
                    length++;
                }
                longest = max(longest , length);
            }
        }
        return longest;
        
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        box = {}
        length = 0 
        l = 0
        for r in range(len(s)):
            if s[r] in box:
                l = max(l , box[s[r]] + 1)
            box[s[r]] = r
            length = max(length , r - l + 1)
        return length

        